src/pluginmain.ts

```diff
@@ -511,6 +511,33 @@ export class MyPlugin {
         });
         infoWindow.open(this._map);
         this._partitions[sheetId].infoWindow = infoWindow;
+
+
+        const path = polygon.getPath();
+        let snappingPath = false
+
+        // when a polygon is edited, compare the changed point to all other
+        // points on the map and set it to the nearest if it's near enough
+        path.addListener('set_at', (index: number) => {
+            if (snappingPath) { return; }
+
+            const zoom: number = this._map.getZoom();
+
+            snappingPath = true;
+            const updatedPoint = path.getAt(index);
+            const nearest = this.getNearestPoint({ polygon, index });
+
+            if (nearest.point &&
+                nearest.distance < 30 &&
```

Comment:
What's the unit on 30?

Reply:
`getNearestPoint()` uses [`google.maps.geometry.spherical.computeDistanceBetween`](https://developers.google.com/maps/documentation/javascript/reference/geometry#spherical.computeDistanceBetween) to get distance in meters.

Reply:
The zoom on the map currently isn't accounted for, so when the user zooms out, the points would have to be positioned visually closer together than when zoomed in.